app/commands/training_data/code_tags_sample/update_tags.rb

```diff
@@ -0,0 +1,31 @@
+class TrainingData::CodeTagsSample::UpdateTags
+  include Mandate
+
+  initialize_with :sample, :tags, :user
+
+  def call
+    ActiveRecord::Base.transaction do
+      sample.lock!
+
+      raise TrainingDataCodeTagsSampleLockedByAnotherUserError unless sample.lockable_by?(user)
```

Comment:
Also, I don't get why this LOC is here at all.

At this stage the record should have been locked when they were editing it, so I'm not sure we need to check it again. What's your thinking?